ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/BufferedWriteChannel.java

```diff
@@ -91,23 +98,26 @@ void flush() throws IOException {
     }
   }
 
-  CompletableFuture<Void> asyncFlush(ExecutorService executor) throws IOException {
+  CompletableFuture<Long> asyncFlush(ExecutorService executor, Long writtenIndex) throws IOException {
     flushBuffer();
     if (forced) {
       return CompletableFuture.completedFuture(null);
     }
-    final CompletableFuture<Void> f = CompletableFuture.supplyAsync(this::fileChannelForce, executor);
+    final CompletableFuture<Long> f =  new CompletableFuture();
+    executor.execute(() -> fileChannelForce(writtenIndex, f));
     forced = true;
+
     return f;
   }
 
-  private Void fileChannelForce() {
+  private void fileChannelForce(Long lastWrittenIndex, CompletableFuture future) {
```

Comment:
Similarly, let's don't pass `lastWrittenIndex` and `future`.  A standard way is to use ` CompletableFuture.supplyAsync(..)`.